src/server/pmix_server_get.c

```diff
@@ -494,7 +526,7 @@ pmix_status_t pmix_pending_resolve(pmix_nspace_t *nptr, int rank,
             /* run through all the requests to this rank */
             PMIX_LIST_FOREACH(req, &lcd->loc_reqs, pmix_dmdx_request_t) {
                 pmix_status_t rc;
-                rc = _satisfy_request(nptr, rank, req->cbfunc, req->cbdata, NULL);
+                rc = _satisfy_request(nptr, rank, NULL, req->cbfunc, req->cbdata, NULL);
                 if( PMIX_SUCCESS != rc ){
```

Comment:
@rhc54 why we omitting the server caddy here?
Does this works fine for the following scenario:
1. client from nspace1 asks for rank=1 from nspace2.
2. Server doesn't have this info and issues direct modex request
3. Server receives dmdx response and calls `pmix_pending_resolve()`
it seems that in this case we won't provide the nspace2 info to the client.